compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -982,12 +984,26 @@ parserInfo numProcessors =
         ])
     )
 
+cliArgsFromDamlYaml :: IO [String]
+cliArgsFromDamlYaml = do
+    handle (\(_ :: ConfigError) -> return [])
+           $ do
+               project <- readProjectConfig $ ProjectPath "."
+               case parseProjectConfig project of
+                   Left _ -> return []
+                   Right pkgConfig -> case cliOpts pkgConfig of
+                       Nothing -> return []
+                       Just xs -> return xs
+
 main :: IO ()
 main = do
     -- We need this to ensure that logs are flushed on SIGTERM.
     installSignalHandlers
     numProcessors <- getNumProcessors
-    withProgName "damlc" $ join $ execParserLax (parserInfo numProcessors)
+    cliArgs <- Env.getArgs
+    damlYamlArgs <- cliArgsFromDamlYaml
```

Comment:
I wonder if we really want to allow all arguments here. The issue with that is that `daml build` accepts different options than `daml ide` so you can end up creating a `daml.yaml` that works with `daml build` but not with `daml ide`. In #2676 I suggested only allowing `--ghc-option` initially which avoids this issue and I would probably also make this logic specific to `execIde` and `execBuild` so we avoid issue with commands that don’t accept `--ghc-option`.

Reply:
That all seems a lot harder to grasp as a mental model for users. "Add these to all invocations of damlc" seems plain enough, and matches the name `damlc-options`; "Add these to only damlc ide and damlc build, and it only supports a subbset of the options they support, and we may change how this behaves in the future" would seem a bit harder for me as a user. I would personally prefer the simple, straightforward behaviour even if it sometimes mean I end up cutting myself.

Reply:
I’m fine with allowing arbitrary flags and expecting users to get this right but I believe that we do at least need to limit the commands affected by this.
There are some commands that do not support `--ghc-options` (e.g. `damlc visualize`) so you would literally be unable to have a `daml.yaml` that sets `--ghc-options` for `damlc build` and `damlc ide` while not breaking `damlc visualize`.